pwiz_tools/Skyline/EditUI/PasteDlg.cs

```diff
@@ -407,8 +414,16 @@ private SrmDocument AddPeptides(SrmDocument document, bool validating, ref Ident
                     }
                 }
                 // Create node using ModificationMatcher.
-                nodePepNew = matcher.GetModifiedNode(pepModSequence, fastaSequence).ChangeSettings(document.Settings,
-                                                                                                  SrmSettingsDiff.ALL);
+                {
+                    nodePepNew = matcher.GetModifiedNode(pepModSequence, fastaSequence);
+                    var settings = document.Settings;
+                    if (!adduct.IsEmpty)
+                    {
+                        settings = settings.ChangeTransitionFilter(f =>
+                            f.ChangePeptidePrecursorCharges(new[] {adduct}));
+                    }
+                    nodePepNew = nodePepNew.ChangeSettings(settings, SrmSettingsDiff.ALL);

```

Comment:
It seems like this would give you a peptide node with auto-select set to true, even when its precursor does not match the true settings. I think you may need to check that the document settings are not already just the adduct (e.g. peptide is charge 2 and the settings are only charge 2). Then if that is not the case, you need to change the settings and remember to set auto-select to false on the peptide.